### PR TITLE
Allow lifting mvpolynomials over non-fields using singular

### PR DESCRIPTION
**Pull request by alexjbest**
It seems singular now supports this, so pass this functionality on to users

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/rings/polynomial/multi_polynomial_libsingular.pyx [hunk 1]
```diff
@@ -4493,6 +4493,27 @@ cdef class MPolynomial_libsingular(MPolynomial_libsingular_base):
...
+            sage: f = I.gen(0) + x2*I.gen(1)
+            sage: f.lift(I)
+            [1, x2]
+            sage: (f+1).lift(I)
+            Traceback (most recent call last):
+            ...
+            ValueError: polynomial is not in the ideal
+            sage: f.lift(I)
+            [1, x2]
+            sage: A.<x,y> = PolynomialRing(ZZ,2,order='degrevlex')
```
Comment (kwankyu) [inline src/sage/rings/polynomial/multi_polynomial_libsingular.pyx:4505]: `f.lift(I)` twice?

File: src/sage/matrix/matrix0.pyx [hunk 2]
```diff
@@ -5813,10 +5813,11 @@ cdef class Matrix(sage.structure.element.Matrix):
             Traceback (most recent call last):
             ...
             ArithmeticError: self must be a square matrix
+
             sage: matrix(RR, 1, 1, [2]).inverse_of_unit()                               # needs sage.libs.singular
             Traceback (most recent call last):
             ...
-            NotImplementedError: Lifting of multivariate polynomials over non-fields is not implemented.
+            ArithmeticError: non-invertible matrix
```
Comment (kwankyu) [inline src/sage/matrix/matrix0.pyx:5821]: I get 
```
sage: M = matrix(RR, 1, 1, [2])
sage: M.is_invertible()
True
sage: M.inverse_of_unit()
[0.500000000000000]
```
as of sage 10.2.beta6.

Comment (alexjbest) [reply src/sage/matrix/matrix0.pyx:5821]: With the definition of RR as in the tests here? I agree it is a confusing set up

File: src/sage/rings/polynomial/multi_polynomial_libsingular.pyx [hunk 3]
```diff
@@ -4493,6 +4493,27 @@ cdef class MPolynomial_libsingular(MPolynomial_libsingular_base):
             sage: f.lift(I)
             [1, x2]
```
Comment (alexjbest) [inline src/sage/rings/polynomial/multi_polynomial_libsingular.pyx]:
```suggestion
```

File: src/sage/rings/polynomial/multi_polynomial_libsingular.pyx [hunk 4]
```diff
@@ -4493,6 +4493,27 @@ cdef class MPolynomial_libsingular(MPolynomial_libsingular_base):
...
+            sage: I = R.ideal(x2**2 + x1 - 2, x1**2 - 1)
+            sage: f = I.gen(0) + x2*I.gen(1)
+            sage: f.lift(I)
+            [1, x2]
+            sage: (f+1).lift(I)
+            Traceback (most recent call last):
+            ...
+            ValueError: polynomial is not in the ideal
+            sage: f.lift(I)
+            [1, x2]
```
Comment (alexjbest) [inline src/sage/rings/polynomial/multi_polynomial_libsingular.pyx]:
```suggestion
```

File: src/sage/rings/polynomial/multi_polynomial_libsingular.pyx (hunk 1 above)
Comment (alexjbest) [reply src/sage/rings/polynomial/multi_polynomial_libsingular.pyx:4505]: Thanks I copied these from above but didn't notice

File: src/sage/matrix/matrix0.pyx (hunk 2 above)
Comment (kwankyu) [reply src/sage/matrix/matrix0.pyx:5821]: Ah, I missed the definitions that come before.

OK.

**Review by kwankyu — APPROVED** [review]
LGTM.

**Pull request merged**